net/raft/raft_test.go

```diff
@@ -141,6 +141,63 @@ func TestClusterSetup(t *testing.T) {
 	}
 }
 
+func TestNodeRestart(t *testing.T) {
+	ctx := context.Background()
+
+	// Create new test cluster
+	nodeA, nodeB, nodeC := newTestCluster(ctx, t)
+	defer nodeA.cleanup()
+	defer nodeB.cleanup()
+	defer nodeC.cleanup()
+
+	httpClient := nodeA.service.client
+
+	_, err := nodeA.service.Exec(ctx, set("/foo", "bar"))
+	must(t, err)
+
+	// Stop the node
+	nodeA.service.Stop()
+
+	// Restart the node
+	nodeA.service, err = Start(nodeA.addr, nodeA.dir, httpClient, nodeA.state)
```

Comment:
Awesome — just changed to `newTestState()` and restart is working without needing to trigger a snapshot